### PR TITLE
SAF IDT support

### PR DESCRIPTION
**Pull request by vit-tomica**
## Proposed changes

This PR adds SAF IDT support to Zowe, specifically to ZSS and ZIS. As APIML is getting ready to leverage SAF IDT tokens (https://github.com/zowe/api-layer/pull/1714), ZSS/ZIS components are the most suitable candidates to be an open source SAF IDT provider for APIML. More info on APIML SAF IDT support:
https://github.com/zowe/docs-site/blob/docs-staging/docs/extend/extend-apiml/implement-new-saf-provider.md

Changes in ZIS: the RACROUTE call is enhanced so that it accepts a new IDTA parameter. A new SAF IDT token is provided to ZSS client.

Changes in ZSS: A new service is created so that it provides two REST endpoints to generate and verify SAF IDT tokens. These endpoints are designed according to the [APIML specifications](https://app.swaggerhub.com/apis/balhar-jakub/saf-idt/1.0.0#/). 

This PR depends upon the following PR in zowe/zowe-common-c: https://github.com/zowe/zowe-common-c/pull/257

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [x] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [x] (I hope) My code follows the style guidelines of this project (see: [Contributing guideline] (https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made
- [x] Relevant update to CHANGELOG.md
- [x] (I hope) My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

## Testing

Tests were created in mocha framework to test the new functionality via REST endpoints locally but are not contributed because I think a repository for ZSS tests does not exists yet or I'm not aware of it. 
The feature has been tested against all three major security managers - RACF, TSS, ACF2

## Further comments

Probably, a new doc should be created so that it helps a user to configure SAF IDT on z/OS installation.

### CODE REVIEW

**Comment by vit-tomica** [comment]
Apologies for PR without early notice.

File: build/build_zss.sh [hunk 1]
```diff
@@ -147,6 +147,7 @@ if c89 \
   ${ZSS}/c/rasService.c \
   ${ZSS}/c/userInfoService.c \
   ${ZSS}/c/passTicketService.c \
+  ${ZSS}/c/safIdtService.c \
```
Comment (lchudinov) [inline build/build_zss.sh:150]: @vit-tomica please include `safIdtService.c` into `build_zss64.sh`, too.

Comment (vit-tomica) [reply build/build_zss.sh:150]: Done.

**Comment by vit-tomica** [comment]
Hi @lchudinov, I was asked by APIML team whether the PR can fit into version 1.27 which has a code freeze this Friday. Thanks!

**Review by lchudinov — APPROVED** [review]
@vit-tomica it looks good to me. @1000TurquoisePogs please decide if we can include this work into 1.27.

**Comment by jalel01** [comment]
@1000TurquoisePogs Hi Sean, ok to include this to 1.27?

**Pull request merged**